docs/concepts/configuration.md

```diff
@@ -51,8 +51,45 @@ As can be seen here, only build-time and deploy-time configuration parameter can
 
 ### Node.js Environment Variables
 
-When running the application in Angular Universal mode within a _Node.js_ environment, we can additionally access the process environment variables via _process.env._ This method provides a way to configure the application at deploy time, e.g., when using docker images.
-Configuration can then be consumed and passed to the client side via means of state transfer.
+When running the application in Angular Universal mode within a _Node.js_ environment, we can additionally access the process environment variables via _process.env_.
+This method provides a way to configure the application at deploy time, e.g., when using docker images.
+Configuration can then be consumed and passed to the client side via means of state transfer using Angulars [TransferState](https://angular.io/api/platform-browser/TransferState).
+
+To introduce a new `TransferState` key it should be added to the `src/app/core/configurations/state-keys.ts`.
+
+```typescript
+export const NEW_KEY = makeStateKey<string>('newKey');
+```
+
+The actual transfer is handled by the `src/app/app.server.module.ts` where the mapping of the environment variable is done.
+
+```typescript
+import { NEW_KEY } from 'ish-core/configurations/state-keys';
+
+export class AppServerModule {
+  constructor(transferState: TransferState) {
+    transferState.set(NEW_KEY, process.env.NEW_KEY);
+  }
+}
+```
+
+Accessing the transferred state in a service or a component is pretty straight forward.
+
+```typescript
+import { NEW_KEY } from 'ish-core/configurations/state-keys';
+
+constructor(
+  private transferState: TransferState,
+) {}
+
+const newKey = this.transferState.get<string>(NEW_KEY, 'default value');
```

Comment:
Shouldn't this be wrapped by `isPlatformBrowser(this.platformId)`?

Reply:
I do not see that this is a general requirement https://angular.io/api/platform-browser/TransferState.
The `get` method should work with the `ServerTransferStateModule` as well.
It might be useful in many situations to wrap it but it probably should not be a general recommendation?

Reply:
You're right, it is not a general requirement, but it should be one for the PWA. If on the server a set comes before the get, then you'll always have a value and `'default value'` is obsolete. The default value would then only be used with webpack-dev-server, which is not a production environment.